lib/kittikrb/cursor.rb

```diff
@@ -146,6 +146,57 @@ def reset_tty!
         self
       end
 
+      # Erase the specified region.
+      # The region describes the rectangle shape which need to erase.
+      def erase(x1, y1, x2, y2)
+        move_to x1, y1
+        length = x2.floor - x1.floor + 1
```

Comment:
First of all, it looks like very cool optimization :+1: But does it really so? Let's explore it together.

I should to say, that we can't move our cursor in this method at all. Simply, we don't have any needs to do this, really. Method `write` moves cursor, so we can't use it. Possibly, it is a problem from architectural view, and we should implement method like `write` that will operate with buffer without moving cursor, but it is not easy question (though it seems so).

Now we have only such thing like `@buffer`. And we can directly operate with it.
We can define our shape as `2, 3, 6, 7`. Or `2, 3, -6, -7`. In the last case we should normalize coordinates first than we start to erase our shape. But it is a corner case. So we can omit it now and fix it later. But. We should write our code with thinking about this problem. So we should implement this method as simple as we can.

Maybe, you heard it before, but 'premature optimization is the root of all evil' (Donald Khuth). Yes, your idea is great. But not now. It makes this code harder to rewrite in the future.

So now you should focus on simplicity of your code. On practice it means that you should use such structure:

```
        y1.floor.upto(y2.floor) do |y|
          x1.floor.upto(x2.floor) do |x|
            
          end
        end
```

Ok. And how we should write to the buffer? It is very easy too.

```
@buffer[get_pointer_from_xy(x, y)] = self.class.wrap ' ', { x: x, y: y }
```

These three lines will be enough for now :)